code/validation/covid19.py

```diff
@@ -81,86 +60,96 @@ def validate_quantile_csv_file(csv_fp, mode, country):
-    else:  # 'wk ahead' in target
-        # NB: we convert `weekdays()` (Monday is 0 and Sunday is 6) to a Sunday-based numbering to get the math to work:
-        weekday_to_sun_based = {i: i + 2 if i != 6 else 1 for i in range(7)}  # Sun=1, Mon=2, ..., Sat=7
-        # 2/4) for x week ahead targets, weekday(target_end_date) should be a Sat
-        if weekday_to_sun_based[target_end_date.weekday()] != 7:  # Sat
-            error_messages.append(f"target_end_date was not a Saturday: {target_end_date}. row={row}")
-            return error_messages  # terminate - remaining validation depends on valid target_end_date
-
-        # set exp_target_end_date and then validate it
-        weekday_diff = datetime.timedelta(days=(abs(weekday_to_sun_based[target_end_date.weekday()] -
-                                                    weekday_to_sun_based[forecast_date.weekday()])))
-        if weekday_to_sun_based[forecast_date.weekday()] <= 2:  # Sun or Mon
-            # 3/4) (Sun or Mon) for x week ahead targets, ensure that the 1-week ahead forecast is for the next Sat
-            delta_days = weekday_diff + datetime.timedelta(days=(7 * (step_ahead_increment - 1)))
-            exp_target_end_date = forecast_date + delta_days
-        else:  # Tue through Sat
-            # 4/4) (Tue on) for x week ahead targets, ensures that the 1-week ahead forecast is for the Sat after next
-            delta_days = weekday_diff + datetime.timedelta(days=(7 * step_ahead_increment))
-            exp_target_end_date = forecast_date + delta_days
-        if target_end_date != exp_target_end_date:
-            error_messages.append(f"target_end_date was not the expected Saturday. forecast_date={forecast_date}, "
-                                  f"target_end_date={target_end_date}. exp_target_end_date={exp_target_end_date}, "
+        error_messages.append(f"Error > non-integer number of weeks ahead in 'wk ahead' target: {target!r}. row={row}")
+        return error_messages  # terminate - depends on valid step_ahead_increment
+
+    # 5. Validate date alignment (Sunday-Saturday epi week)
+    weekday_to_sun_based = {i: i + 2 if i != 6 else 1 for i in range(7)}  # Sun=1, Mon=2, ..., Sat=7
+
+    # 5.1 for x week ahead targets, weekday(target_end_date) should be a Sat
+    if weekday_to_sun_based[target_end_date.weekday()] != 7:
+       error_messages.append(f"target_end_date was not a Saturday: {target_end_date}. row={row}")
+       return error_messages  # terminate - depends on valid target_end_date
+   
+    # 5.2 Forecast date should always be Mon
+    if abs(weekday_to_sun_based[forecast_date.weekday()]) != 2:
+            error_messages.append(f"Error > forecast_date was not a Monday, row={row}")
+
+    # 5.3 For x week ahead targets, ensure 1-week ahead forecast is for next Sat
+    #   - set exp_target_end_date and then validate it
+    weekday_diff = datetime.timedelta(days=(abs(weekday_to_sun_based[target_end_date.weekday()] -
```

Comment:
Doesn't need the `abs`, right? We already checked that `weekday_to_sun_based[target_end_date.weekday()]` is 7 so this should be >= 0